### DISCUSSION TITLE
IPNS Keygen from Signatures and Identifiers

### DISCUSSION THREAD

**Discussion by sshmatrix** [namesys-eth/ccip2-eth-resources#10] [Announcements]
ed25519 keygen is successfully tested here: 

https://github.com/dostr-eth/nostr-tools/tree/ipns

I will use this for IPNS keygen in our client

**Comment by 0xc0de4c0ffee** [comment]
ed25519 is default.. can we use secp256k1? so we can reuse same keys for IPNS and record signer?

**Reply by sshmatrix** [reply]
We can use `secp256k1` in IPNS now? I didn't see it in the docs. I figured it is still using RSA and `ed25519`. I will have a look. Issue might be w3name package, which I believe handles only `ed25519` keys (need to doublecheck this though)

**Reply by sshmatrix** [reply]
> ed25519 is default.. can we use secp256k1? so we can reuse same keys for IPNS and record signer?

<img width="748" alt="Screenshot 2023-04-17 at 16 54 03" src="https://user-images.githubusercontent.com/19473027/232470787-28ab4dee-f40f-4dc0-8143-9aff7f066139.png">

^ w3name only supports `ed25519` for now. It seems trivial to bypass this default setting but I wouldn't want to distract myself with it right now. We are not handling the private keys manually anywhere, so the user will likely never know/care whether `ed25519` is used or `secp256k1`

**Reply by 0xc0de4c0ffee** [reply]
w3name can act as primary IPNS provider but we also need backup pubsub+ipns and any other web2/3 features services to propagate IPNS updates in few minutes/seconds delay for real dynamic upgrades.. + last line of backup is domain.eth owners running their own IPFS node to import/manage their IPNS records. Keys should be exportable in IPFS client's import/export formats.
